x-pack/plugins/security_solution/public/common/containers/alerts/use_alert_prevalence_from_process_tree.ts

```diff
@@ -36,20 +46,129 @@ interface ProcessTreeAlertPrevalenceResponse {
   alertIds: string[];
 }
 
-export function useAlertPrevalenceFromProcessTreeActual(
-  processEntityId: string
+interface EntityResponse {
+  id: string;
+  name: string;
+  schema: object;
+}
+
+interface TreeResponse {
+  statsNodes: Array<{
+    data: object;
+    id: string;
+    parent: string;
+    stats: {
+      total: number;
+      byCategory: {
+        alerts?: number;
+      };
+    };
+  }>;
+  alertIds: string[];
+}
+
+function useAlertDocumentAnalyzerSchema(processEntityId: string) {
+  const http = useHttp();
+  const query = useQuery<EntityResponse[]>(['getAlertPrevalenceSchema', processEntityId], () => {
+    return http.get<EntityResponse[]>(`/api/endpoint/resolver/entity`, {
+      query: {
+        _id: processEntityId,
+        indices: ['.alerts-security.alerts-default', 'logs-*'],
+      },
+    });
+  });
+  if (query.isLoading) {
+    return {
+      loading: true,
+      error: false,
+      id: null,
+      schema: null,
+    };
+  } else if (query.data) {
+    const {
+      data: [{ schema, id }],
+    } = query;
+    return {
+      loading: false,
+      error: false,
+      id,
+      schema,
+    };
+  } else {
+    return {
+      loading: false,
+      error: true,
+      id: null,
+      schema: null,
+    };
+  }
+}
+
+export function useAlertPrevalenceFromProcessTree(
+  processEntityId: string,
+  timelineId: string
 ): UserAlertPrevalenceFromProcessTreeResult {
   const http = useHttp();
+  const getStartSelector = useMemo(() => startSelector(), []);
```

Comment:
this is all duplicated code from graph_overlay, going to create a new hook to share these selectors etc